contracts/evmscript/executors/DelegateScript.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity 0.4.18;
+
+import "../ScriptHelpers.sol";
+import "../IEVMScriptExecutor.sol";
+
+
+interface DelegateScriptTarget {
+    function exec() public;
+}
+
+
+contract DelegateScript is IEVMScriptExecutor {
+    using ScriptHelpers for *;
+
+    uint256 constant internal SCRIPT_START_LOCATION = 4;
+
+    /**
+    * @notice Executes script by delegatecall into a contract
+    * @param script [ specId (uint32 = 2) ][ contract address (20 bytes) ]
+    * @param input ABI encoded call to be made to contract (if empty executes default exec() function)
+    * @param blacklist If any address is passed, will revert.
+    * @return Call return data
+    */
+    function execScript(bytes script, bytes input, address[] blacklist) external returns (bytes) {
+        require(blacklist.length == 0); // dont have ability to control bans, so fail.
+
+        // Script should be spec id + address (20 bytes)
+        require(script.length == SCRIPT_START_LOCATION + 20);
+        return delegate(script.addressAt(SCRIPT_START_LOCATION), input);
+    }
+
+    /**
+    * @dev Delegatecall to contract with input data
+    */
+    function delegate(address addr, bytes memory input) internal returns (bytes memory output) {
+        require(isContract(addr));
+        require(addr.delegatecall(input.length > 0 ? input : defaultInput()));
+        return returnedData();
+    }
+
+    function isContract(address _target) internal view returns (bool) {
```

Comment:
We should make a `ChecksContract` base contract with this and have `DelegateProxy` and this contract inherit from it.